CRM/ManualDirectDebit/Hook/Custom/MandateDataGenerator.php

```diff
@@ -6,71 +6,70 @@
 class CRM_ManualDirectDebit_Hook_Custom_MandateDataGenerator {
 
   /**
-   * Primary ID of Direct Debit Mandate
+   * Direct Debit Mandate table name
    *
-   * @var int
+   * @var string
    */
-  private $mandateId;
+  const MANDATE_TABLE_NAME = 'civicrm_value_dd_mandate';
 
   /**
-   * Contact entity ID
+   * List of the mandate fields to be generated
    *
-   * @var int
+   * @var array
    */
-  private $entityID;
+  private $fieldsToGenerate = [
+    'dd_ref' => FALSE,
+    'start_date' => FALSE,
+  ];
 
   /**
-   * Parameters which submitted by form
+   * Array of extension settings
    *
    * @var array
    */
-  private $savedFields;
+  private $settings;
 
   /**
-   * List of the mandate fields to be generated
+   * Parameters which submitted by form
    *
    * @var array
    */
-  private $fieldsToGenerate = [
-    'collection_day' => FALSE,
-    'dd_ref' => FALSE,
-    'authorisation_date' => FALSE,
-    'start_date' => FALSE,
-  ];
+  private $savedFields;
 
   /**
-   * Array of extension settings
+   * Primary ID of Direct Debit Mandate
    *
-   * @var array
+   * @var int
    */
-  private $settings;
+  private $mandateId;
 
-  public function __construct($entityID, &$params) {
-    $this->entityID = $entityID;
+  public function __construct($entityID, $settings, &$params) {
+    $this->settings = $settings;
     $this->savedFields = $params;
-    $this->setMandateId();
-    $this->setManualDirectDebitSettings();
+    $this->setMandateId($entityID);
```

Comment:
For consistency and clarity, it might be better to just do:
```php
$this->settings = $settings;
$this->savedFields = $params;
$this->mandateId = $this->getLastMandateId($entityID);
```

Please check if we actually need the setter function for mandate ID, I believe it is only used here. Or if you feel we should use setter functions, then shouldn't we also use them for $settings and $savedFields?